upstairs/src/upstairs.rs

```diff
@@ -1119,8 +1115,7 @@ impl Upstairs {
          * ID and the next_id are connected here, in that all future writes
          * should be flushed at the next flush ID.
          */
-        let mut gw = self.guest.guest_work.lock().await;
-        let gw_id = gw.next_gw_id();
+        let gw_id = self.guest.guest_work.next_gw_id();
```

Comment:
The guest work lock (along with the former downstairs lock) provided protection
during IO submission that prevented any interleaving of IO requests with each other.

In the new "one task" world, there will only ever be one task that calls
`submit_flush`
`submit_read`
`submit_write`
Correct? 

Is this also enforced by the way the new `Guest` struct is split, such that only one owner can exist and attempt to handle incoming IOs?

Reply:
It has always been possible to clone a `Guest` and submit many jobs simultaneously, e.g. Propolis uses 8x Crucible worker tasks.

In the old code, jobs were serialized through the `VecDeque` in the `Guest`, which was locked by both sides (8x worker tasks and one Upstairs task).  Now, jobs are serialized through an `mpsc` queue between the `Guest` and the `GuestIoHandle`, which should be equivalent.

`Upstairs::submit_flush/read/write` all take `&mut self`, so they are guaranteed to only be called by the single owner of the `struct Upstairs`.

Reply:
> Upstairs::submit_flush/read/write all take &mut self, so they are guaranteed to only be called by the single owner of the struct Upstairs.

Perfect, this is what I wanted to see/hear